cclib/parser/psi4parser.py

```diff
@@ -40,6 +40,17 @@ def before_parsing(self):
         # with changes triggered by ==> things like this <== (Psi3 does not have this)
         self.section = None
 
+        # Match the number of skipped lines required based on the type
+        # of gradient present (determined from the header), as
+        # otherwise the parsing is identical.
+        self.gradient_headers = {

```

Comment:
Why not make this one dictionary with analytic/numerical as the keys, or perhaps a namedtuple? I don't think we need to care about the performance here, so looping through a list/keys is fine.